spring-boot/admin/src/test/java/eu/coatrack/admin/controllers/PublicAPIAuthenticationAndAuthorizationViaHeaderTokenTest.java

```diff
@@ -0,0 +1,66 @@
+package eu.coatrack.admin.controllers;
+
+import eu.coatrack.admin.security.CustomUserInfoTokenServices;
+import eu.coatrack.api.ServiceApiDTO;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.oauth2.provider.OAuth2Authentication;
+import org.springframework.security.oauth2.provider.OAuth2Request;
+import org.springframework.web.client.RestTemplate;
+
+import java.util.Collections;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.when;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.DEFINED_PORT)
+public class PublicAPIAuthenticationAndAuthorizationViaHeaderTokenTest {
+
+    private static final String AUTHORIZATION_TOKEN = "testToken";
+    private static final String PUBLIC_API_URL = "http://localhost:8080/public-api/services";
+
+    @Autowired
+    private RestTemplate restTemplate;
+
+    @MockBean
+    private CustomUserInfoTokenServices customUserInfoTokenServicesMock;
+
+    //test to call the public api and get the services
+    @Test
+    public void CallThePublicAPIWithAuthorizationBearerToken_ShouldRetrieveListOfServices() {
+
+        UsernamePasswordAuthenticationToken userAuthentication = new UsernamePasswordAuthenticationToken("exampleCompanyInc",
+                "test", Collections.singleton(new SimpleGrantedAuthority("ROLE_USER")));
```

Comment:
* Instead of "test" it should be more descriptive like "examplePassword" or "testPassword".
* Also please create an extra variable for the authority object and pass it as argument. Logic as argument should be avoided.
* "exampleCompanyInc" is used twice in this class. Please create a variable like "companyName" for abstraction.

Reply:
So exampleCompanyInc is used as it is the default username for tests on the DatabaseInitializer. So I indeed I created a variable as the suggestion here is good, but I called it `DEFAULT_USERNAME_FOR_TESTING`.
Regarding the `test`, as the parameter was called credentials, I renamed test to `testCredentials`.
Regarding the argument with the SimpleGrantedAuthority, I also created a variable as you suggested.
So this is done